lib/streamlit/runtime/state/common.py

```diff
@@ -232,23 +236,67 @@ def failure(
 ]
 
 
+def _register_element_id(element_type: str, element_id: str) -> None:
+    """Register the element ID and key for the given element.
+
+    If the element ID or key is not unique, an error is raised.
+
+    Parameters
+    ----------
+
+    element_type : str
+        The type of the element to register.
+
+    element_id : str
+        The ID of the element to register.
+
+    Raises
+    ------
+
+    StreamlitDuplicateElementKey
+        If the element key is not unique.
+
+    StreamlitDuplicateElementID
+        If the element ID is not unique.
+
+    """
+    from streamlit.runtime.scriptrunner import get_script_run_ctx
```

Comment:
I think we want to avoid inner-function imports anymore except for lazy-loading

Reply:
Hmm, unfortunately, this causes a circular import :( And I don't really like adding it as a parameter ... I will take a look if it can be moved somewhere to prevent the circular import.